### ISSUE TITLE
Modifications done in foreach-by-reference are ignored

### ISSUE THREAD

**Issue by miken32** [phpstan/phpstan-strict-rules#261]
PHPStan complains about this code at level 5, thinking I'm passing an array of strings instead of an array of integers. This _seems_ like a bug, but I'm pretty new to this stuff, so apologies if it is obviously not!

```php
<?php

/**
 * @param array<int> $nums
 */
function firstNum(array $nums): int{
	return $nums[0] ?? 0;
}

$arr = ["f", "o", "o"];
foreach ($arr as &$v) {
    $v = ord($v);
}
unset($v);
echo firstNum($arr);
// output: 102
```

I was thinking this was similar behaviour to how PHPStan won't enter into methods to see what's happening. But I can create another variable, assign the result of `ord()` to it, and it doesn't complain. So clearly it is following the code into the `foreach` loop and knows that integers are being put into the array.
```php
$final = [];
foreach ($arr as $v) {
    $final[] = ord($v);
}
```

https://phpstan.org/r/9d39def4-9aad-4492-9dbc-bb082b839517

**Comment by miken32** [comment]
oops, wrong repo!
